### PR TITLE
[14.0][MIG] purchase_manual_currency

### PR DESCRIPTION
**Pull request by PrapassornS**

### CODE REVIEW

File: purchase_manual_currency/README.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+.. contents::
+   :local:
+
+Usage
+=====
+
+To use this module, the company have to access rights Multi Currencies.
+
+Step to used manual currency:
+#. Go to Invoicing > Configuration > Settings > Currencies > Multi-Currencies
```
Comment (kittiu) [inline purchase_manual_currency/README.rst:47]: Need to spare one line. So .rst will show correctly.

```suggestion

#. Go to Invoicing > Configuration > Settings > Currencies > Multi-Currencies
```

File: purchase_manual_currency/models/purchase.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    @api.onchange("currency_id", "date_order")
+    def _onchange_currency_change_rate(self):
+        today = fields.Date.today()
+        main_currency = self.env.company.currency_id
+        ctx = {"company_id": self.company_id.id, "date": self.date_order or today}
+        self.custom_rate = main_currency.with_context(**ctx)._get_conversion_rate(
+            main_currency, self.currency_id, self.company_id, self.date_order or today
+        )
+
+    def action_refrese_currency(self):
```
Comment (kittiu) [inline purchase_manual_currency/models/purchase.py]: Miss spell?
```suggestion
    def action_refresh_currency(self):
```

File: purchase_manual_currency/models/purchase.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        self.ensure_one()
+        if self.state != "draft":
+            raise ValidationError(_("Rate currency can refrese state draft only."))
```
Comment (kittiu) [inline purchase_manual_currency/models/purchase.py]:
```suggestion
            raise ValidationError(_("Rate currency can refresh state draft only."))
```

File: purchase_manual_currency/views/purchase_views.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                        attrs="{'invisible': [('is_manual', '=', False)], 'readonly': [('state', '!=', 'draft')]}"
+                    />
+                    <field
+                        name="custom_rate"
+                        attrs="{'invisible': [('manual_currency', '=', False)], 'readonly': [('state', '!=', 'draft')]}"
+                    />
+                    <button
+                        aria-label="Opened"
+                        type="object"
+                        name="action_refrese_currency"
```
Comment (kittiu) [inline purchase_manual_currency/views/purchase_views.xml]:
```suggestion
                        name="action_refresh_currency"
```

File: purchase_manual_currency/models/purchase.py (hunk 2 above)
Comment (kittiu) [reply purchase_manual_currency/models/purchase.py]: @Saran440 misspell or intentional? Fix should be in v13 also?

Comment (Saran440) [reply purchase_manual_currency/models/purchase.py]: yes, I typo. and agree with you that it should be fix in v13.

**Comment by kittiu** [comment]
@PrapassornS I think it will be easier to fix the labels error into v13 first. And then you can migrate it into v14. WDYT?
(So far, I don't see a lot of code changes in from v13 -> v14)

**Review by kittiu — APPROVED** [review]

**Review by Saran440 — APPROVED** [review]
No major change

File: purchase_manual_currency/__manifest__.py [hunk 5]
```diff
@@ -2,7 +2,7 @@
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 {
     "name": "Purchase - Manual Currency",
-    "version": "13.0.1.0.1",
+    "version": "14.0.1.0.1",
```
Comment (HviorForgeFlow) [inline purchase_manual_currency/__manifest__.py]: it should be 14.0.1.0.0

**Review by HviorForgeFlow — COMMENTED** [review]
Just a nitpicking.. :smile:

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
@PrapassornS

File: purchase_manual_currency/__manifest__.py (hunk 5 above)
Comment (PrapassornS) [reply purchase_manual_currency/__manifest__.py]: Thanks, I will do that

**Comment by HviorForgeFlow** [comment]
Check travis errors, not sure maybe it can be greenify using assertAlmostEqual() unittest function.

**Review by HviorForgeFlow — APPROVED** [review]

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
See https://github.com/OCA/purchase-workflow/pull/1069

**Review by HviorForgeFlow — APPROVED** [review]
:partying_face:

**Pull request merged**